wire-ios-request-strategy/Sources/Message Sending/MessageSenderTests.swift

```diff
@@ -0,0 +1,356 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+
+final class MessageSenderTests: MessagingTestBase {
+
+    func testThatWhenSecurityLevelIsDegraded_thenFailWithSecurityLevelDegraded() async throws {
+        // given
+        syncMOC.performAndWait {
+            groupConversation?.setPrimitiveValue(
+                NSNumber(value: ZMConversationSecurityLevel.secureWithIgnored.rawValue), forKey: "securityLevel"
+            )
+        }
+
+        let message = GenericMessageEntity(
+            conversation: groupConversation,
+            message: GenericMessage(content: Text(content: "Hello World")),
+            completionHandler: nil)
+
+        let (_, messageSender) = Arrangement(coreDataStack: coreDataStack)
```

Comment:
might be good to keep the tests consistent with the rest of code base, using `sut` so we know what we test
```suggestion
        let (_, sut) = Arrangement(coreDataStack: coreDataStack)
```

Reply:
🤔

Reply:
I'll do this at end since otherwise I'll get a ton of git conflicts